File: backend/app/models/category.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey
from sqlalchemy.orm import relationship

from app.core.database import Base


class Category(Base):
    """
    Fully database-driven — admin can add/edit/deactivate/reorder without
    any code change (requirement #13). No categories are hard-coded here;
    they get inserted via a seed script or the admin API.
    """
    __tablename__ = "categories"

    id = Column(Integer, primary_key=True, index=True)
    name_hi = Column(String(100), nullable=False)
    name_en = Column(String(100), nullable=False)
    icon_url = Column(String(255), nullable=True)
    sort_order = Column(Integer, default=0)
    is_active = Column(Boolean, default=True)

    products = relationship("Product", back_populates="category")

File: seller_app/lib/models/category.dart
class Category {
  final int id;
  final String nameHi;
  final String nameEn;
  final String? iconUrl;
  final int sortOrder;
  final bool isActive;

  Category({
    required this.id,
    required this.nameHi,
    required this.nameEn,
    this.iconUrl,
    this.sortOrder = 0,
    this.isActive = true,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      nameHi: json['name_hi']?.toString() ?? '',
      nameEn: json['name_en']?.toString() ?? '',
      iconUrl: json['icon_url']?.toString(),
      sortOrder: json['sort_order'] ?? 0,
      isActive: json['is_active'] ?? true,
    );
  }

  String displayName(String lang) => lang == 'hi' ? nameHi : nameEn;
}
